src/main/java/com/terraformersmc/terrestria/feature/trees/BryceTreeFeature.java

```diff
@@ -7,25 +7,21 @@
 import com.terraformersmc.terrestria.feature.trees.components.SmallRoots;
 import com.terraformersmc.terrestria.feature.trees.templates.SmallLogTree;
 import net.minecraft.block.BlockState;
+import net.minecraft.tag.BlockTags;
 import net.minecraft.util.math.BlockBox;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
 import net.minecraft.world.ModifiableTestableWorld;
 import net.minecraft.world.TestableWorld;
-import net.minecraft.world.World;
 import net.minecraft.world.gen.feature.BranchedTreeFeatureConfig;
-import net.minecraft.world.gen.feature.DefaultFeatureConfig;
 import net.minecraft.world.gen.feature.TreeFeatureConfig;
 
-import java.util.ArrayList;
 import java.util.Random;
 import java.util.Set;
 import java.util.function.Function;
 
 public class BryceTreeFeature extends SmallLogTree implements Branches, SmallRoots {
 
-	private ArrayList<BlockPos> leafOrigins = new ArrayList<>();
-
 	public BryceTreeFeature(Function<Dynamic<?>, ? extends BranchedTreeFeatureConfig> configDeserializer, boolean updateNeighbor, BlockState log, BlockState leaves) {
```

Comment:
We should use the state providers within the tree feature config instead of passing `log` and `leaves` parameters.

Reply:
This I believe would be an issue for a seperate PR in the future, as it's an issue with all trees.